src/ILCompiler.TypeSystem/tests/ILTestAssembly/Signature.il

```diff
@@ -0,0 +1,86 @@
+.class private auto ansi beforefieldinit Atom
+       extends [CoreTestAssembly]System.Object
+{
+  .method public hidebysig specialname rtspecialname 
+          instance void  .ctor() cil managed
+  {
+    // Code size       8 (0x8)
+    .maxstack  8
+    IL_0000:  ldarg.0
+    IL_0001:  call       instance void [CoreTestAssembly]System.Object::.ctor()
+    IL_0006:  nop
+    IL_0007:  ret
+  } // end of method I::.ctor
+
+} // end of class Atom
+
+.class private auto ansi beforefieldinit A`1<U>
+       extends [CoreTestAssembly]System.Object
+{
+  .method public hidebysig specialname rtspecialname 
+          instance void  .ctor() cil managed
+  {
+    // Code size       8 (0x8)
+    .maxstack  8
+    IL_0000:  ldarg.0
+    IL_0001:  call       instance void [CoreTestAssembly]System.Object::.ctor()
+    IL_0006:  nop
+    IL_0007:  ret
+  } // end of method A`1::.ctor
+
+} // end of class A`1
+
+.class private auto ansi beforefieldinit BaseClass`2<U,T>
```

Comment:
This passes even without your fix. You really ought to be writing a test that needs your fix to behave correctly. Try adding more Method methods that have different mod opts before and after one of these, and only the right method should match.